bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/launch/LaunchUtils.java

```diff
@@ -122,10 +125,31 @@ private static void loadConfigurations(Map<String, String> configArgs, String co
                 ((BLogManager) logManager).loadUserProvidedLogConfiguration();
             }
 
-            boolean observeFlag = ConfigRegistry.getInstance().getAsBoolean(CONFIG_OBSERVABILITY_ENABLED);
-            if (observeFlag) {
-                ConfigRegistry.getInstance().addConfiguration(CONFIG_METRICS_ENABLED, Boolean.TRUE);
-                ConfigRegistry.getInstance().addConfiguration(CONFIG_TRACING_ENABLED, Boolean.TRUE);
+            if (ConfigRegistry.getInstance().contains(CONFIG_OBSERVABILITY_ENABLED)) {
+                // Only try to inherit if "b7a.observability" config is provided
+                boolean observeFlag = ConfigRegistry.getInstance().getAsBoolean(CONFIG_OBSERVABILITY_ENABLED);
```

Comment:
I think this is always `true`

Reply:
Can the key exist and value be `false` in config registry ?
A user can set it and that is why I changed it to this way from previous implementation.

Reply:
Ahh got it, thanks.